Add environments cleanup to KEB (#1428)
kyma-incubator/compass: docs/kyma-environment-broker/03-07-environments-cleanup.md

# Environments Cleanup

## Overview

> This application cleans up environments which do not meet requirements in given Gardener project.

Reviewer: ```suggestion
Environments Cleanup is an application that cleans up environments that do not meet requirements in a given Gardener project.
```

## Prerequisites

Environments Cleanup requires access to:

> 1. Gardener Project of choice to filter Shoots without proper label

Reviewer: ```suggestion
- Gardener project of choice to filter Shoots without a proper label
```

> 2. Database to get Instance ID for each Runtime marked for deletion

Reviewer: ```suggestion
- Database to get an Instance ID for each Runtime marked for deletion
```

> 3. Kyma Environment Broker to trigger Runtime deprovisioning

Reviewer: ```suggestion
- Kyma Environment Broker to trigger Runtime deprovisioning
```

## Configuration

The Environments Cleanup binary allows you to override some configuration parameters. You can specify the following environment variables:

| Environment variable                       | Description                                                                                                                        | Default value                                                            |
|--------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------|

> | APP_MAX_AGE_HOURS                          | Defines the maximum time a Shoot can live without deletion in case the label is not specified. The Shoot age is provided in hours. | `24h`                                                                    |

Reviewer: Please put all variables' names in **bold**.

| APP_LABEL_SELECTOR                          | Defines the label selector to filter out Shoots for deletion. | `owner.do-not-delete!=true`                                                                    |

> | APP_GARDENER_PROJECT                       | Specifies Gardener project name.                                                                                                   | `kyma-dev`                                                               |

Reviewer: ```suggestion
| APP_GARDENER_PROJECT                       | Specifies the name of a Gardener project.                                                                                                   | `kyma-dev`                                                               |
```

> | APP_GARDENER_KUBECONFIG_PATH               | Specifies Gardener cluster's kubeconfig path.                                                                                      | `/gardener/kubeconfig/kubeconfig`                                        |

Reviewer: ```suggestion
| APP_GARDENER_KUBECONFIG_PATH               | Specifies the kubeconfig path to a Gardener cluster.                                                                                      | `/gardener/kubeconfig/kubeconfig`                                        |
```

> | APP_DATABASE_USER | Database username | `postgres` |

Reviewer: ```suggestion
| APP_DATABASE_USER | Specifies the username for the database. | `postgres` |
```

> | APP_DATABASE_PASSWORD | Database user password | `password` |

Reviewer: ```suggestion
| APP_DATABASE_PASSWORD | Specifies the user password for the database. | `password` |
```

> | APP_DATABASE_HOST | Database host | `localhost` |

Reviewer: ```suggestion
| APP_DATABASE_HOST | Specifies the host of the database. | `localhost` |
```

> | APP_DATABASE_PORT | Database port | `5432` |

Reviewer: ```suggestion
| APP_DATABASE_PORT | Specifies the port for the database. | `5432` |
```

> | APP_DATABASE_NAME | Database name | `provisioner` |

Reviewer: ```suggestion
| APP_DATABASE_NAME | Specifies the name of the database. | `provisioner` |
```

> | APP_DATABASE_SSL_MODE | SSL Mode for PostgrSQL. See all the possible values [here](https://www.postgresql.org/docs/9.1/libpq-ssl.html)  | `disable`|

Reviewer: ```suggestion
| APP_DATABASE_SSL_MODE | Activates the SSL mode for PostgrSQL. See all the possible values [here](https://www.postgresql.org/docs/9.1/libpq-ssl.html).  | `disable`|
```
